modules/i18n/src/main/java/com/devonfw/module/i18n/common/util/I18nUtils.java

```diff
@@ -0,0 +1,198 @@
+      throws JSONException {
+
+    String closingBraces = I18nConstants.EMPTY_STRING;
+    String strJSON = I18nConstants.EMPTY_STRING;
+    if (filter.contains(I18nConstants.DOT)) {
+      int indexOfDot = filter.indexOf(I18nConstants.DOT);
+      String subKey = filter.substring(0, indexOfDot);
+      JsonObject jsonObject = (JsonObject) objJSON.get(subKey);
+      count++;
+      strJSONKey = strJSONKey + "{\"" + subKey + "\":";
+      if (jsonObject == null) {
+        throw new JSONException(subKey + " is null");
+      }
+      try {
+        return getWithDotNotation(jsonObject, filter.substring(indexOfDot + 1), strJSONKey, count);
+      } catch (JSONException e) {
+        throw new JSONException(subKey + I18nConstants.DOT + e.getMessage());
+      }
+    } else {
+      for (int i = 0; i <= count; i++) {
+        closingBraces = closingBraces + I18nConstants.CLOSING_BRACE;
+      }
+
+      if (objJSON.get(filter) == null)
+        strJSON = I18nConstants.OPENING_BRACE + I18nConstants.CLOSING_BRACE;
+      else
+        strJSON = strJSONKey + "{\"" + filter + "\":" + objJSON.get(filter).toString() + closingBraces;
+    }
+    return strJSON;
+  }
+
+  /**
+   * @param objJSON is the JSON representation of the resource file
+   * @param filter is the filter supplied from service
+   * @return JSON object filtered by the filter
+   * @throws Exception thrown by getJsonObj
+   */
+
+  public static String getJsonObj(JsonObject objJSON, String filter) throws Exception {
+
+    String strJSON = null;
+    String strJSONKey = I18nConstants.EMPTY_STRING;
+    int count = I18nConstants.ZERO;
+    if (filter == null) {
+      throw new JSONException("Null key.");
+    }
+    if (objJSON != null) {
+      strJSON = getWithDotNotation(objJSON, filter, strJSONKey, count);
+    }
+    return strJSON;
+  }
+
+  /**
+   * @param resourceMap contains the resources in the form of key value pairs
+   * @param filter received from service
+   * @return resources as JSON String
+   * @throws Exception thrown by getResourcesAsJSON
+   */
+  public static String getResourcesAsJSON(HashMap<String, String> resourceMap, String filter) throws Exception {
+
+    JsonParser jsonParser = new JsonParser();
+
+    final java.lang.reflect.Type mapType = new TypeToken<Map<String, String>>() {
+    }.getType();
+
+    String strJSON = new GsonBuilder().registerTypeAdapter(mapType, new BundleMapSerializer()).create()
+        .toJson(resourceMap, mapType);
+
+    if (filter != null && !filter.isEmpty()) {
+      JsonObject obj = (JsonObject) jsonParser.parse(strJSON);
+      strJSON = I18nUtils.getJsonObj(obj, filter);
+    }
+    return strJSON;
+  }
+
+  /**
+   * @param locale received from service
+   * @return resources generated by MMM as Map
+   * @throws Exception thrown by getResourcesGeneratedFromMMMAsMap
+   */
+
+  public static Map<String, String> getResourcesGeneratedFromMMMAsMap(Locale locale) throws Exception {
```

Comment:
Avoid `CAPITALIZED` segments in names but use `camlCaseAlsoForAbbreviationsLikeMmm`.
https://github.com/devonfw/devon4j/blob/develop/documentation/coding-conventions.asciidoc#naming